server/datastore/mysql/queries.go

```diff
@@ -198,8 +221,26 @@ func (ds *Datastore) NewQuery(
 }
 
 // SaveQuery saves changes to a Query.
-func (ds *Datastore) SaveQuery(ctx context.Context, q *fleet.Query) error {
-	sql := `
+func (ds *Datastore) SaveQuery(ctx context.Context, q *fleet.Query, shouldDiscardResults bool) error {
```

Comment:
Just in case, similar to `ApplyQueries`, let's define the error as a "named return":
```
func (ds *Datastore) SaveQuery(ctx context.Context, q *fleet.Query, shouldDiscardResults bool) (err error) {
```

Reply:
@lucasmrod done!